pkgs/by-name/ou/outfly/package.nix

```diff
@@ -0,0 +1,98 @@
+{
+  lib,
+  fetchFromGitea,
+  buildType ? "release",
+  platform ? "linux",
+  enableWayland ? true,
+  enableXorg ? true,
+  enableOpenGL ? true,
+  enableVulkan ? true,
+  rustPlatform,
+  makeDesktopItem,
+  pkg-config,
+  libxkbcommon,
+  alsa-lib,
+  libGL,
+  vulkan-loader,
+  wayland,
+  libXrandr,
+  libXcursor,
+  libX11,
+  libXi,
+  ...
+}:
+
+assert (platform == "linux") -> (enableWayland || enableXorg);
+assert (platform == "linux") -> (enableVulkan || enableOpenGL);
+
+rustPlatform.buildRustPackage rec {
+  pname = "outfly";
+  inherit buildType;
+  version = "0.13.0";
+  src = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "outfly";
+    repo = "outfly";
+    rev = "v${version}";
+    hash = "sha256-A43lLG5K9mEhRsv+paQa1so4Cyk75vyocAyU3k6wDiQ=";
+  };
+
+  buildNoDefaultFeatures = true;
+  buildFeatures = [
+    "embed_assets"
+  ] ++ lib.optionals enableXorg [ "x11" ] ++ lib.optionals enableWayland [ "wayland" ];
+
+  runtimeInputs =
+    [ libxkbcommon ]
+    ++ lib.optionals enableOpenGL [ libGL ]
+    ++ lib.optionals enableXorg [
+      libXrandr
+      libX11
+    ]
+    ++ lib.optionals enableVulkan [ vulkan-loader ];
+
+  buildInputs = [
+    alsa-lib.dev
+    libXcursor
+    libXi
+  ] ++ lib.optionals enableWayland [ wayland ];
+
+  nativeBuildInputs = [ pkg-config ];
+  doCheck = false;
+
+  postFixup = ''
+    patchelf $out/bin/outfly \
+    --add-rpath ${lib.makeLibraryPath runtimeInputs}
+  '';
```

Comment:
Unfortunately the paths are stripped in fixup. iirc cargo opens external dependencies generally using dlopen, and since that is not detected by fixup, those paths are marked as unnecessary